src/analysis/special_functions/arsinh.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2020 James Arthur. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: James Arthur, Chris Hughes, Shing Tak Lam
+-/
+import analysis.special_functions.trigonometric
+
+open real
+noncomputable theory
+
+/-!
+# Inverse of the sinh function
+
+In this file we prove that sinh is bijective and hence has an
+inverse, arsinh.
+
+## Main Results
+
+- `sinh_injective`: The proof that `sinh` is injective
+- `sinh_surjective`: The proof that `sinh` is surjective
+- `sinh_bijective`: The proof `sinh` is bijective
+- `arsinh`: The inverse function of `sinh`
+
+## Tags
+
+arsinh, arcsinh, argsinh, asinh, sinh injective, sinh bijective, sinh surjective
+-/
+
+
+/-- `arsinh` is defined using a logarithm, `arsinh x = log (x + sqrt(1 + x^2))`. -/
+def arsinh (x : ℝ) := log (x + sqrt (1 + x^2))
+
+/-- `sinh` is injective, `∀ a b, sinh a = sinh b → a = b`. -/
+lemma sinh_injective : function.injective sinh := sinh_strict_mono.injective
+
+private lemma aux_lemma (x : ℝ) : 1 / (x + sqrt (1 + x ^ 2)) = -x + sqrt (1 + x ^ 2) :=
+begin
+  refine (eq_one_div_of_mul_eq_one _).symm,
+  have : 0 ≤ 1 + x ^ 2 := add_nonneg zero_le_one (pow_two_nonneg x),
+  rw [add_comm, ← sub_eq_neg_add, ← mul_self_sub_mul_self,
+      mul_self_sqrt this, pow_two, add_sub_cancel]
+end
+
+private lemma b_lt_sqrt_b_one_add_sq (b : ℝ) : b < sqrt (1 + b ^ 2) :=
+calc  b
+    ≤ sqrt (b ^ 2)     : le_sqrt_of_sqr_le $ le_refl _
+... < sqrt (1 + b ^ 2) : (sqrt_lt (pow_two_nonneg _) (add_nonneg zero_le_one (pow_two_nonneg _))).2
+  (lt_one_add _)
+
+private lemma add_sqrt_one_add_pow_two_pos (b : ℝ) : 0 < b + sqrt (1 + b ^ 2) :=
+by { rw [← neg_neg b, ← sub_eq_neg_add, sub_pos, pow_two, neg_mul_neg, ← pow_two],
+  exact b_lt_sqrt_b_one_add_sq (-b) }
+
+/-- `arsinh` is the right inverse of `sinh`. -/
+lemma arsinh_sinh (x : ℝ) : sinh (arsinh x) = x :=
```

Comment:
Aren't this name and `sinh_arsinh` swapped?

Reply:
I've just sorted this... this is very embarrassing